Add property page

zfregin opened the pull request:
Add property page component within a protected route. Contains Formik form with Yup validation to post form data to /properties api. Things to review include:
- [x] Styling
- [ ] Validation
- [ ] Successful POST

Still need to review the need to adjust the backend endpoint to accept units as a parameter. Will it accept an assigned property manager in addition to this?

zfregin committed:
Formik form and submit

Initial set up for Formik form and submit handler

zfregin committed:
User context and POST authorization

Set up the context to enable authorization for POST request

zfregin committed:
Styling and OnBlur removal

Added CSS styling, removed the OnBlur error styling and messages. Created a Button primitive as well but needs approval.

zfregin committed:
Removed Button component and classnames package

zfregin committed:
Update _addproperty.scss

zfregin committed:
Validation Error messages

Chris-Boe commented on dwellingly/src/App.js:
```diff
@@ -87,6 +88,7 @@ export class App extends React.Component {
               <PrivateRoute exact path='/' component={Dashboard} />
               <Route exact path='/login' component={LoginForm} />
               <PrivateRoute exact path='/dashboard' component={Dashboard} />
+              <Route exact path='/add-property' component={AddProperty} />
```
Since we should have persistent auth merged in soon, we should make this a "PrivateRoute" since you need to be authenticated to use it

Chris-Boe commented on dwellingly/src/views/addProperty.js:
```diff
@@ -0,0 +1,162 @@
+                                                placeholder="10"
+                                            />
+                                        </div>
+                                        <div className=" add-property__assign-manager-container">
```
Like we talked about, let's comment this out and make a new issue for it. Be sure to mention the `react-select` package in the issue.
It'll be waiting on #120 to get done, then we can start using that api endpoint.

Chris-Boe commented on dwellingly/src/views/addProperty.js:
```diff
@@ -0,0 +1,162 @@
+const validationSchema = Yup.object().shape({
+    name: Yup.string()
+        .min(5, "*Name must have at least 2 characters")
+        .max(50, "*Names can't be longer than 100 characters")
```
Getting mixed signals here 😄 Sounds like the min should be 2 and the max should be 100

Chris-Boe commented on dwellingly/src/views/addProperty.js:
```diff
@@ -0,0 +1,162 @@
+    zipcode: Yup.string()
+        .required("*Zipcode is required"),
+    state: Yup.string()
+        .required("*State is required")
```
We'll want to keep in mind in the future a general "Address schema" that we can check forms against to make sure we have common types for things like state (OR or Oregon?), Zip (5 digits? or 5 digits + the postal extension?), etc.
Not sure what that will look like yet so we don't need to worry about it here

Chris-Boe reviewed (changes requested):
Nice and sleek!

Chris-Boe commented on dwellingly/src/views/addProperty.js:
```diff
@@ -0,0 +1,162 @@
+                                                type="text"
+                                                name="units"
+                                                onChange={handleChange}
+                                                value={values.zipcode}
```
Looks like this line has a copy/paste issue.
I noticed when I type in the 'zip' form element, it adds my input to the "Unit" box, and I can't type into the Unit box directly.

![image](https://user-images.githubusercontent.com/8600792/78210602-8462bd00-745f-11ea-8c3e-0ac0122fa5fb.png)

Chris-Boe commented on dwellingly/src/views/addProperty.js:
```diff
@@ -0,0 +1,162 @@
+});
+
+const formHandler = (data, context) => {
+    axios.post('http://localhost:5000/properties', data, { headers: {"Authorization" : `Bearer ${context.user.accessJwt}`} })
```
This _looks_ right but I'm getting a 422 with an error saying it's missing the Bearer token. I think "Authorization" should _not_ be in quotes by what I can find online

Chris-Boe commented on dwellingly/src/views/addProperty.js:
```diff
@@ -0,0 +1,162 @@
+const formHandler = (data, context) => {
+    axios.post('http://localhost:5000/properties', data, { headers: {"Authorization" : `Bearer ${context.user.accessJwt}`} })
+        .then(function(response){
+            console.log(response);
```
We'll want to make a Toast or something for both the OnSuccess handling and the Onfailure handling, but for now let's switch them to alert(response) so we can see what goes on

zfregin committed:
PR Fixes

Validation schema fixed. Units field and account manager elements removed for later implementation

zfregin committed:
Merge branch 'development' into add-property-page

zfregin committed:
Alert Message

Chris-Boe commented on dwellingly/src/App.js:
```diff
@@ -90,6 +91,7 @@ export class App extends React.Component {
               <PrivateRoute exact path='/' component={Dashboard} />
               <Route exact path='/login' component={LoginForm} />
               <PrivateRoute exact path='/dashboard' component={Dashboard} />
+              <PrivateRoute exact path='/add-property' component={AddProperty} />
```
We have the `/add/property` PrivateRoute in the list below from the left nav, so we won't need this one

Chris-Boe commented on dwellingly/src/views/addProperty.js:
```diff
@@ -0,0 +1,172 @@
+        .max(50, "*City can't be longer than 50 characters")
+        .required("*City is required"),
+    zipcode: Yup.string()
+        .max(50, "*City can't be longer than 20 characters")
```
Couple of copy/paste errors in here and the state validation below

zfregin committed:
PR fixes
